src/main/java/frc/robot/subsystems/BuddyBalanceSubsystem.java

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.util.datalog.DataLog;
+import edu.wpi.first.util.datalog.DoubleLogEntry;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.hardware.MotorController;
+import frc.robot.hardware.MotorController.MotorConfig;
+import frc.robot.classes.TunableNumber;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxPIDController;
+
+public class BuddyBalanceSubsystem extends SubsystemBase {
+  // public static double kDockedPosition; // Docked position may or may not be used depending on final design of buddy balance
+  public static double kBalancedPosition; // Buddy balance PID reference point when lifting a robot and engaging charge station
+  public static double kDeployedPosition; // Buddy balance PID reference point when setting down a robot/initial position when deployed
+  // TODO: Make the reference point constants final when they are done being tuned with TunableNumbers
+  public static final double kDefaultMotorP = 0.000001;
+  public static final double kDefaultMotorI = 0;
+  public static final double kDefaultMotorD = 0.000001;
+  private TunableNumber refPointDockedTuner;
+  private TunableNumber refPointBalancedTuner;
+  private TunableNumber refPointDeployedTuner;
+
+  private CANSparkMax rightMotor;
+  private CANSparkMax leftMotor;
+  private SparkMaxPIDController rightPIDController;
+  private SparkMaxPIDController leftPIDController;
+  private RelativeEncoder rightEncoder;
+  private RelativeEncoder leftEncoder;
+  private DataLog datalog = DataLogManager.getLog();
+  private DoubleLogEntry buddyBalancePosition = new DoubleLogEntry(datalog, "/buddybalance/position");
+  private ShuffleboardTab buddyBalanceTab = Shuffleboard.getTab("Buddy Balance"); // TODO: Replace buddy balance tab with whatever tab the position should be logged to
+  private GenericEntry positionEntry;
+  private boolean isDeployed = false;
+
+  private TunableNumber tNumBuddyBalanceRightP;
+  private TunableNumber tNumBuddyBalanceRightI;
+  private TunableNumber tNumBuddyBalanceRightD;
+  private TunableNumber tNumBuddyBalanceLeftP;
+  private TunableNumber tNumBuddyBalanceLeftI;
+  private TunableNumber tNumBuddyBalanceLeftD;
```

Comment:
Don't need to specify BuddyBalance, it's already in the subsystem
For naming purposes, it might be better to suffix tunable numbers with Tuner, instead of prefixing with tNum